packages/back-end/src/models/ProjectModel.ts

```diff
@@ -1,126 +1,82 @@
-import mongoose from "mongoose";
-import uniqid from "uniqid";
 import { DEFAULT_STATS_ENGINE } from "shared/constants";
-import { omit } from "lodash";
 import { hasReadAccess } from "shared/permissions";
+import { z } from "zod";
 import { ApiProject } from "../../types/openapi";
 import { ProjectInterface, ProjectSettings } from "../../types/project";
-import { ReqContext } from "../../types/organization";
-import { ApiReqContext } from "../../types/api";
+import { StatsEngine } from "../../types/stats";
+import { tuple } from "../util/types";
+import { baseSchema, MakeModelClass, Context } from "./BaseModel";
 
-const projectSchema = new mongoose.Schema({
-  id: {
-    type: String,
-    unique: true,
-  },
-  organization: {
-    type: String,
-    index: true,
+const engineTypes = tuple<StatsEngine>()(["bayesian", "frequentist"]);
+
+const projectValidator = baseSchema
+  .extend({
+    name: z.string(),
+    description: z.string(),
+    settings: z.object({
+      statsEngine: z.enum(engineTypes),
+    }),
+  })
+  .strict();
+
+const BaseClass = MakeModelClass({
+  schema: projectValidator,
+  collectionName: "projects",
+  idPrefix: "prj__",
+  auditLog: {
+    entity: "project",
+    createEvent: "project.create",
+    updateEvent: "project.update",
+    deleteEvent: "project.delete",
   },
-  name: String,
-  description: String,
-  dateCreated: Date,
-  dateUpdated: Date,
-  settings: {},
+  projectScoping: "none",
+  globallyUniqueIds: true,
 });
 
-type ProjectDocument = mongoose.Document & ProjectInterface;
+export class ProjectModel extends BaseClass {
+  protected canRead(doc: ProjectInterface) {
+    return hasReadAccess(this.context.readAccessFilter, doc.id);
+  }
+  protected canCreate() {
+    return true;
+  }
+  protected canUpdate() {
+    return true;
+  }
+  protected canDelete() {
+    return true;
+  }
+}
 
-const ProjectModel = mongoose.model<ProjectInterface>("Project", projectSchema);
+export const findAllProjectsByOrganization = (context: Context) =>
+  new ProjectModel(context).getAll();
```

Comment:
I'm not the biggest fan of this pattern. I think I'd prefer not exporting helper methods here.  Instead, add this class to `context.models` and access the methods like this throughout the code.

```ts
await context.models.projects.getAll();
```